jina/drivers/cache.py

```diff
@@ -21,23 +21,20 @@ class BaseCacheDriver(BaseIndexDriver):
     def __init__(self, with_serialization: bool = False, *args, **kwargs):
         self.with_serialization = with_serialization
         super().__init__(*args, **kwargs)
-        self.field = None
 
     def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
-        self.field = self.exec.field
-
         if self._method_name == 'update':
-            self.exec_fn([d.id for d in docs], [d.id if self.field == ID_KEY else d.content_hash for d in docs])
+            values = [''.join(d.get_attrs(*self.exec.fields).values()) for d in docs]
+            self.exec_fn([d.id for d in docs],
+                         values)
         else:
             for d in docs:
-                data = d.id
-                if self.field == CONTENT_HASH_KEY:
-                    data = d.content_hash
-                result = self.exec[data]
+                value = ''.join(d.get_attrs(*self.exec.fields).values())
```

Comment:
having a cache value like this can cause unwanted collisions.
Example:
```
d1.a = '123'
d1.b = '456'
d2.a = '1'
d2.b = '23456'
```

both documents would have the same cache value

Reply:
I would suggest creating a hash value from the list

Reply:
You can create the hash using:
```
hash(('123', '456'))
-3037693913774805154
hash(('1', '23456'))
-8391814287327595382
```

Reply:
Also the values should not be converted to strings since the conversion changes the hash value
```
hash(('1', 23456))
8044249239569849110
hash(('1', '23456'))
-8391814287327595382
```

Reply:
Good idea. I adapted the logic to use `hash`. Theoretically this can still fail if 

```
d1.a = '123'
d1.c = '456'
d2.a = '123'
d2.b = '456'
```